build/parseSpec.c

```diff
@@ -501,17 +511,23 @@ int readLine(rpmSpec spec, int strip)
 	goto retry;
     }
 
-after_classification:
-    if (match != -1) {
+    if (lineType->id & (LINE_IFANY)) {
 	rl = xmalloc(sizeof(*rl));
 	rl->reading = spec->readStack->reading && match;
 	rl->next = spec->readStack;
 	rl->lineNum = ofi->lineNum;
+	rl->readable = (!rl->reading) && (spec->readStack->reading);
 	rl->lastConditional = lineType;
 	spec->readStack = rl;
 	spec->line[0] = '\0';
+    } else if (lineType->id & (LINE_ELIF | LINE_ELIFARCH | LINE_ELIFOS)) {
```

Comment:
The various ELIF's are another candidate for a bit mask, this set is repeated at least twice in the patch (used earlier in the "dont expand in false branch" check). In general, when addressing something, it pays to look for similar cases/opportunities/bugs, they're quite often there.

Reply:
It is changed in the new version.